internal/clients/team4/trust.go

```diff
@@ -0,0 +1,72 @@
+package team4
+
+import "github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+
+type trust struct {
+	trustMap map[shared.ClientID]float64
+}
+
+func (t *trust) GetTrustMatrix() []float64 {
+	trustMatrix := make([]float64, len(t.trustMap))
+	for clientID, trustValue := range t.trustMap {
+		index := int(clientID)
+		if index < len(trustMatrix) {
+			trustMatrix[index] = trustValue
+		}
+	}
+	return trustMatrix
+}
+
+func (t *trust) GetClientTrust(clientID shared.ClientID) float64 {
+
+	if clientTrust, ok := t.trustMap[clientID]; ok {
+		return clientTrust
+	}
+	return 0
+}
+
+func (t *trust) ChangeClientTrust(clientID shared.ClientID, diff float64) {
+	// diff is percentage to change trust for clientID ie. diff in range [-1,1]
+	if _, ok := t.trustMap[clientID]; ok {
+		t.trustMap[clientID] = t.trustMap[clientID] * (1 + diff)
+		t.normalise()
+	}
+}
+
+func (t *trust) SetClientTrust(clientID shared.ClientID, newValue float64) {
+	// diff is percentage to change trust for clientID ie. diff in range [-1,1]
+	if _, ok := t.trustMap[clientID]; ok {
+		t.trustMap[clientID] = newValue
+		t.normalise()
+	}
+}
+
+func (t *trust) totalTrustSum() float64 {
+	totalTrust := 0.0
+	for _, trust := range t.trustMap {
+		totalTrust += trust
+	}
+	return totalTrust
+}
+
+func (t *trust) expectedTrustSum() float64 {
+	return 0.5 * float64(len(t.trustMap))
+}
+
+//normalise ensures that trust values are always in range [0,1]
+func (t *trust) normalise() {
+	// diff is percentage to increase trust for clientID ie. diff in range [0,1]
+	if len(t.trustMap) > 0 {
+		normaliseCoef := t.expectedTrustSum() / t.totalTrustSum()
```

Comment:
divide by 0 check, if you trust no one

Reply:
yeah, although we maintain the average trust at 0.5, so this situation will not occur. I'll add a check for the peace of mind